salt/modules/boto_secgroup.py

```diff
@@ -221,15 +242,22 @@ def get_group_id(name, vpc_id=None, region=None, key=None, keyid=None, profile=N
     '''
     conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
 
-    group = _get_group(conn, name, vpc_id, region)
+    if not vpc_id and vpc_name:
```

Comment:
This logic is duplicated in a bunch of places. It could probably be a part of _get_group, if you modified _get_group to take a vpc_name. The logic in _get_group would lookup the group in the vpc if a vpc_id was passed in, or if a vpc_name was passed it, it would lookup the vpc_id first.